app/src/main/java/com/example/babytracking/BreastActivity.kt

```diff
@@ -1,11 +1,136 @@
+import kotlinx.android.synthetic.main.breast_layout.breast_quality_mother
+import kotlinx.android.synthetic.main.breast_layout.spinner_breast
+
+class BreastActivity : AppCompatActivity() {
+
+    private var hour: Int = 0
+    private var minute: Int = 0
+    private var ampm: String = "AM"
+    private var breast: String = ""
 
-class BreastActivity: AppCompatActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.breast_layout)
+
+        breastTimeClick()
+        breastSelection()
+        breastSave()
+    }
+
+    private fun breastTimeClick() {
+        val timePicker = findViewById<TimePicker>(R.id.sleep_timepick_hour)
+        timePicker.setOnTimeChangedListener { _, pickerHour, pickerMinute ->
+            hour = pickerHour
+            minute = pickerMinute
+
+            // AM_PM decider logic
+            when {
+                hour == 0 -> {
+                    hour += 12
+                    ampm = "AM"
+                }
+                hour == 12 -> ampm = "PM"
+                hour > 12 -> {
+                    hour -= 12
+                    ampm = "PM"
+                }
+                else -> ampm = "AM"
+            }
+        }
+    }
+
+    private fun breastSelection() {
+        if (spinner_breast != null) {
+            val breastSpinnerAdapter = ArrayAdapter.createFromResource(
+                this, R.array.array_breast,
+                android.R.layout.simple_spinner_item
+            )
+
+            breastSpinnerAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
+            spinner_breast.adapter = breastSpinnerAdapter
+
+
+            spinner_breast.onItemSelectedListener =
+                object : AdapterView.OnItemSelectedListener {
+                    override fun onItemSelected(
+                        parent: AdapterView<*>, view: View, pos: Int, id: Long
+                    ) {
+                        if (! TextUtils.isEmpty(parent.getItemAtPosition(pos) as String)) {
+                            breast = spinner_breast.selectedItem.toString()
+                        }
+                    }
+
+                    override fun onNothingSelected(parent: AdapterView<*>?) {
+                        breast = getString(R.string.unknown_breast)
+                    }
+                }
+        }
+    }
+
+    private fun breastSave() {
+        breast_button_save.setOnClickListener {
+
+            val lengthText: String = breast_length.text.toString().trim()
+            val lenghtInt: Int = Integer.parseInt(lengthText)
+            val babyText = breast_quality_baby.text.toString().trim()
+            val motherText = breast_quality_mother.text.toString().trim()
+            val hourVal = if (hour < 10) "0" + hour else hour
+            val minVal = if (minute < 10) "0" + minute else minute
+            val timeText = "$hourVal : $minVal $ampm"
```

Comment:
See the part about using a timestamp rather than this. This logic is fine, but should be used only for display. Additionally, there are a ton of utility classes that can format a timestamp to a readable time / date for you, so you don't have to handle that logic yourself